02-express-tutorial/authorize.js

```diff
@@ -0,0 +1,18 @@
+const authorize = (req, res, next) => {
+    const { user } = req.query;
+    if(user === "john"){
+        // http://localhost:5000/?user=john will give you access to communicate with the server. 
+        req.user = { name: "john", id:3 }
+        next()
+    }
+    else{
+        res.status(401).send("Unauthorized");
+    };
```

Comment:
For what it's worth, this code produces an error when the user is unauthorized, because it sends a response but does not stop processing the request. (Because there is no `return`, `next()` is called below and the following error logs:
![image](https://user-images.githubusercontent.com/22882623/215328628-a37c8423-3881-418f-b176-c8ccf3c8ac98.png)

It's a good practice to pretty much always `return res....` to prevent additional processing from accidentally occuring.

Reply:
Excellent!

I will fix this now please. Thanks Taylor!